release.sh

```diff
@@ -111,6 +115,29 @@ function deploy_files_to_repository {
     git push "$GIT_UPSTREAM_URL" "$RELEASE_TAG"
 }
 
+function merge_branches {
+    GIT_UPSTREAM_URL=$("$SCRIPT_DIR/git-upstream-url.py")
+    MAJOR_BRANCHES=($(git branch | tr -d " *" | grep -E "^[0-9]+\.x\.x$") master)
```

Comment:
This only gets local branches, if there were a new major release that wasn't checked out locally it would be missing here.

You need to get (or add) the remote for `$GIT_UPSTREAM_URL`, fetch it, and then get the branches for that remote.